### PR TITLE
add configuration parameter to allow miners to specify their bip135 vote

### PR DESCRIPTION
**Pull request by gandrewstone**

### CODE REVIEW

**Comment by dgenr8** [comment]
Cool

File: qa/rpc-tests/bip68-112-113-p2p.py [hunk 1]
```diff
@@ -209,11 +209,8 @@ def get_tests(self):
         self.tip = int("0x" + self.nodes[0].getbestblockhash(), 0)
         self.nodeaddress = self.nodes[0].getnewaddress()
 
-        assert_equal(get_bip9_status(self.nodes[0], 'csv')['status'], 'defined')
         test_blocks = self.generate_blocks(61, 4)
         yield TestInstance(test_blocks, sync_every_block=True) # 1
-        # Advanced from DEFINED to STARTED, height = 143
-        assert_equal(get_bip9_status(self.nodes[0], 'csv')['status'], 'started')
```
Comment (sickpig) [inline qa/rpc-tests/bip68-112-113-p2p.py:216]: not clear to me why all this `get_bip9` got removed

File: src/consensus/params.h [hunk 2]
```diff
@@ -26,13 +26,13 @@ enum DeploymentPos
...
-    DEPLOYMENT_UNASSIGNED_BIT_1,
-    DEPLOYMENT_UNASSIGNED_BIT_2,
-    DEPLOYMENT_UNASSIGNED_BIT_3,
-    DEPLOYMENT_UNASSIGNED_BIT_4,
-    DEPLOYMENT_UNASSIGNED_BIT_5,
-    DEPLOYMENT_UNASSIGNED_BIT_6,
-    DEPLOYMENT_UNASSIGNED_BIT_7,
+    DEPLOYMENT_128MB_BLOCK,
+    DEPLOYMENT_MUL_SHIFT_INVERT,
+    DEPLOYMENT_UNLIMITED_SCRIPT_SIZE,
```
Comment (sickpig) [inline src/consensus/params.h]: Maybe it's just bike shedding but I would call it something `unlimited number of op_codes` or `remove op-codes limit`

File: src/versionbits.h [hunk 3]
```diff
@@ -84,4 +86,18 @@ ThresholdState VersionBitsState(const CBlockIndex *pindexPrev,
     VersionBitsCache &cache);
 uint32_t VersionBitsMask(const Consensus::Params &params, Consensus::DeploymentPos pos);
 
+// tweak to configure and dynamically change what you are voting for
+extern std::string bip135Vote;
+
+// Set the specified feature to the vote.  Pass vote=-1 to check existence of feature
+// returns false if the feature is not defined.
+bool AssignBip135Vote(const std::string &feature, int vote = true);
+// Set all votes to false
```
Comment (ptschip) [inline src/versionbits.h:96]: I think vote=true is a little confusing and maybe prone to operator error, I was thinking to set vote=-1 if no second param was set, and then return the current status so the operator can know whether it's on or not before trying to change it.  With the way it is now, I'm afraid someone may end up turning on a vote when they didn't intend to.

**Review by sickpig — APPROVED** [review]

File: src/rpc/mining.cpp [hunk 4]
```diff
@@ -530,6 +532,14 @@ returns JSON if pblockOut is NULL
 pblockOut -A copy of the block if not NULL
 */
 
+bool forceTemplateRecalc = false;
+// force block template recalculation
```
Comment (ptschip) [inline src/rpc/mining.cpp:536]: it this needs cs_main then maybe 

+bool forcTemplateRecalc = false GUARDED_BY(cs_main)

File: qa/rpc-tests/bip68-112-113-p2p.py (hunk 1 above)
Comment (gandrewstone) [reply qa/rpc-tests/bip68-112-113-p2p.py:216]: csv is activated a long time ago so this portion of the test seems unnecessary today.  We can't reuse the csv bit (0), until we get it out of the code and that code is no longer being run.

File: src/rpc/mining.cpp (hunk 4 above)
Comment (ptschip) [reply src/rpc/mining.cpp:536]: oops, that will probably give compile errors on mac. should be

+bool forceTemplateRecalc GUARDED_BY(cs_main) = false;

File: src/consensus/params.h (hunk 2 above)
Comment (gandrewstone) [reply src/consensus/params.h]: you are right, the size is still constrained.
DEPLOYMENT_UNLIMITED_SCRIPT_OPCODES,

**Pull request merged**